lib/ToolbarBadgeHub.jsm

```diff
@@ -52,12 +60,23 @@ class _ToolbarBadgeHub {
     // Need to wait for ASRouter to initialize before trying to fetch messages
     await waitForInitialized;
     this.messageRequest("toolbarBadgeUpdate");
+    // Listen for pref changes that could trigger new badges
+    Services.prefs.addObserver(this.prefs.WHATSNEW_TOOLBAR_PANEL, this);
+  }
+
+  observe(aSubject, aTopic, aPrefName) {
+    switch (aPrefName) {
+      case this.prefs.WHATSNEW_TOOLBAR_PANEL:
+        this.messageRequest("toolbarBadgeUpdate");
+        break;
+    }
   }
 
   executeAction({ id }) {
     switch (id) {
       case "show-whatsnew-button":
         ToolbarPanelHub.enableToolbarButton();
+        ToolbarPanelHub.enableAppmenuButton();
```

Comment:
This behaves oddly when using the local panel messages because their targeting is overriden such that the `isWhatsNewPanelEnabled` isn't checked in the targetting anymore. And neither of these two functions are checking the pref. So I am currently in a situation where I have the pref disabled but all the buttons are appearing. This is how I launched:

```
./mach run --setpref 'browser.newtabpage.activity-stream.asrouter.providers.panel_local_testing={"id":"panel_local_testing", "cohort": "SHOW_TEST"}' --setpref 'browser.messaging-system.whatsNewPanel.enabled=false' --setpref 'browser.newtabpage.activity-stream.asrouter.devtoolsEnabled=true'
```